openquake/hazardlib/gsim/projects/acme_2019.py

```diff
@@ -0,0 +1,371 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import math
+import numpy as np
+from openquake.hazardlib import const
+from openquake.hazardlib.gsim.base import GMPE, registry, CoeffsTable
+from openquake.hazardlib.imt import SA
+from openquake.hazardlib.gsim.chiou_youngs_2014 import ChiouYoungs2014
+from openquake.hazardlib.gsim.yenier_atkinson_2015 import \
+        YenierAtkinson2015BSSA
+from openquake.hazardlib.gsim.nga_east import (get_phi_s2ss_at_quantile,
+                                               get_tau_at_quantile,
+                                               get_phi_ss_at_quantile,
+                                               get_phi_ss,
+                                               TAU_SETUP,
+                                               PHI_SETUP,
+                                               PHI_S2SS_MODEL,
+                                               TAU_EXECUTION)
+
+PATH = os.path.join(os.path.dirname(__file__), "..", "nga_east_tables")
+
+
+def get_sof_adjustment(rake, imt):
+    """
+    Computes adjustment factor for style-of-faulting following the scheme
+    proposed by Bommer et al. (2003).
+
+    :param rake:
+        Rake value
+    :param imt:
+        The intensity measure type
+    :return:
+        The adjustment factor
+    """
+    if imt.name == 'PGA' or (imt.name == 'SA' and imt.period <= 0.4):
+        f_r_ss = 1.2
+    elif imt.name == 'SA' and imt.period > 0.4:
+        f_r_ss = 1.2 - (0.3/np.log10(3.0/0.4))*np.log10(imt.period/0.4)
+    else:
+        raise ValueError('Unsupported IMT')
+    # Set coefficients
+    f_n_ss = 0.95
+    p_r = 0.85
+    p_n = 0.0
+    # Normal - F_N:EQ
+    if -135 < rake <= -45:
+        famp = f_r_ss**(-p_r) * f_n_ss**(1-p_n)
+    # Reverse - F_R:EQ
+    elif 45 < rake <= 135:
+        famp = f_r_ss**(1-p_r) * f_n_ss**(-p_n)
+    # Strike-Slip - F_SS:EQ
+    elif ((-30 < rake <= 30) or (150 < rake <= 180) or (-180 < rake <= -150)):
+        famp = f_r_ss**(-p_r) * f_n_ss**(-p_n)
+    else:
+        raise ValueError('Unrecognised rake value')
+    return famp
+
+
+class YenierAtkinson2015ACME2019(YenierAtkinson2015BSSA):
+    """
+    This is a modified version of the
+    :class:`openquake.hazardlib.gsim.yenier_atkinson_2015.YenierAtkinson2015BSSA`
+    which incorporates a correction of the median ground motion based on the
+    focal mechanism.
+    """
+
+    non_general = True
+
+    def __init__(self):
+        # Initialise using the super class
+        super().__init__(focal_depth=10., region='CENA')
+        # Adjust the set of required parameters for the description of the
+        # rupture by adding rake
+        _previous = [s for s in list(super().REQUIRES_RUPTURE_PARAMETERS)]
```

Comment:
Why the list comprehension? `list` is enough.

Reply:
corrected to only use list